estransport/estransport.go

```diff
@@ -86,9 +86,12 @@ func (c *Client) Perform(req *http.Request) (*http.Response, error) {
 	}
 
 	c.setURL(u, req)
-	c.setBasicAuth(u, req)
 	c.setUserAgent(req)
 
+	if _, ok := req.Header["Authorization"]; !ok {
```

Comment:
I suppose it's sensible for request headers to override headers defined based on client/transport configuration... but the auth headers may be based on env vars, which one might expect override headers defined in code? Either way this behaviour should be well documented, it'll probably surprise someone either way you go :)